datajoint/schema.py

```diff
@@ -87,15 +87,18 @@ def process_relation_class(relation_class, context):
 
         process_relation_class(cls, context=self.context)
 
-        # Process subordinate relations
-        parts = list()
-        is_part = lambda x: inspect.isclass(x) and issubclass(x, Part)
+        # Process part relations
+        def is_part(x):
+            return inspect.isclass(x) and issubclass(x, Part)
 
-        for var, part in inspect.getmembers(cls, is_part):
-            parts.append(part)
-            part._master = cls
-            # TODO: look into local namespace for the subclasses
-            process_relation_class(part, context=dict(self.context, **{cls.__name__: cls}))
+        parts = list()
+        for part in dir(cls):
+            if part[0].isupper():
```

Comment:
Yes, we assume that class names are in CamelCase notation, starting with a capital letter and containing only alphanumerical characters.  No underscores.  This is assume for a consistent translation into mysql names.  The translation is performed by `datajoint.utils.from_camel_case`.